### PR TITLE
mobile: Add an engine termination test with open streams

### PR DESCRIPTION
**Pull request by abeyad**
Tested with:
```
bazelisk test --runs_per_test=100 --config=remote --test_output=streamed --cache_test_results=no --test_arg="-l trace" //test/common/integration:client_integration_test 
```

### CODE REVIEW

File: mobile/test/common/integration/client_integration_test.cc [hunk 1]
```diff
@@ -391,13 +400,27 @@ void ClientIntegrationTest::explicitFlowControlWithCancels(uint32_t body_size) {
     } else {
       stream->readData(100);
     }
+
+    if (terminate_engine && request_for_engine_termination == i) {
+      {
```
Comment (RyanTheOptimist) [inline mobile/test/common/integration/client_integration_test.cc:405]: nit: Do we need these '{}'s?

**Review by RyanTheOptimist — APPROVED** [review]
Nice!

File: mobile/test/common/integration/client_integration_test.cc (hunk 1 above)
Comment (abeyad) [reply mobile/test/common/integration/client_integration_test.cc:405]: good point, we don't, I'll send a PR now to fix this.

**Pull request merged**